### ISSUE TITLE
`app run` in a project with a nested main package doesn't work on Windows

### ISSUE THREAD

**Issue by tarple** [restx/restx#180]
app new test-ms

it won't start up, prematurely failing notifying me that the class cannot be found

``` logs
app new test-ms

    found io.restx#restx-monitor-codahale;0.34 in public
    found io.dropwizard.metrics#metrics-core;3.1.0 in public
    found org.slf4j#slf4j-api;1.7.7 in public
    found io.dropwizard.metrics#metrics-healthchecks;3.1.0 in public
    found io.dropwizard.metrics#metrics-jvm;3.1.0 in public
    found io.dropwizard.metrics#metrics-graphite;3.1.0 in public
    found io.restx#restx-monitor-admin;0.34 in public
    found io.restx#restx-log-admin;0.34 in public
    found ch.qos.logback#logback-classic;1.0.13 in public
    found ch.qos.logback#logback-core;1.0.13 in public
    found io.restx#restx-i18n-admin;0.34 in public
    found io.restx#restx-i18n;0.34 in public
    found io.restx#restx-stats-admin;0.34 in public
    found com.github.kevinsawicki#http-request;5.5 in public
    found io.restx#restx-server-jetty;0.34 in public
    found javax.servlet#javax.servlet-api;3.0.1 in public
    found io.restx#restx-servlet;0.34 in public
    found org.codehaus.janino#janino;2.6.1 in public
    found org.codehaus.janino#commons-compiler;2.6.1 in public
    found org.eclipse.jetty#jetty-server;8.1.8.v20121106 in public
    found org.eclipse.jetty.orbit#javax.servlet;3.0.0.v201112011016 in public
    found org.eclipse.jetty#jetty-continuation;8.1.8.v20121106 in public
    found org.eclipse.jetty#jetty-http;8.1.8.v20121106 in public
    found org.eclipse.jetty#jetty-io;8.1.8.v20121106 in public
    found org.eclipse.jetty#jetty-util;8.1.8.v20121106 in public
    found org.eclipse.jetty#jetty-servlet;8.1.8.v20121106 in public
    found org.eclipse.jetty#jetty-security;8.1.8.v20121106 in public
    found org.eclipse.jetty#jetty-webapp;8.1.8.v20121106 in public
    found org.eclipse.jetty#jetty-xml;8.1.8.v20121106 in public
    found org.eclipse.jetty#jetty-servlets;8.1.8.v20121106 in public
    found org.eclipse.jetty#jetty-client;8.1.8.v20121106 in public
    found io.restx#restx-apidocs;0.34 in public
    found com.fasterxml.jackson.module#jackson-module-jsonSchema;2.3.3 in public

    found io.restx#restx-specs-admin;0.34 in public
    found io.restx#restx-specs-tests;0.34 in public
    found junit#junit;4.11 in public
    found org.hamcrest#hamcrest-core;1.3 in public
    found org.assertj#assertj-core;1.6.0 in public
    found com.googlecode.java-diff-utils#diffutils;1.3.0 in public
:: resolution report :: resolve 3642ms :: artifacts dl 37ms
    :: evicted modules:
    com.fasterxml.jackson.core#jackson-annotations;2.3.0 by [com.fasterxml.jacks
on.core#jackson-annotations;2.3.3] in [default, test]
    joda-time#joda-time;2.1 by [joda-time#joda-time;2.3] in [default, test]
    com.google.guava#guava;15.0 by [com.google.guava#guava;17.0] in [default, te
st]
    com.google.guava#guava;11.0.2 by [com.google.guava#guava;17.0] in [default,
test]
    org.slf4j#slf4j-api;1.7.5 by [org.slf4j#slf4j-api;1.7.7] in [default]
    ---------------------------------------------------------------------
    |                  |            modules            ||   artifacts   |
    |       conf       | number| search|dwnlded|evicted|| number|dwnlded|
    ---------------------------------------------------------------------
    |      default     |   77  |   0   |   0   |   5   ||   71  |   0   |
    |      runtime     |   0   |   0   |   0   |   0   ||   0   |   0   |
    |       test       |   32  |   0   |   0   |   4   ||   28  |   0   |
    ---------------------------------------------------------------------

:: problems summary ::
:::: WARNINGS
        module not found: test#test-ui;latest.integration

    ==== local: tried

      C:\Users\pgayroberts\.ivy2\local\test\test-ui\[revision]\ivys
\ivy.xml

      -- artifact test#test-ui;latest.integration!test-ui.jar:

      C:\Users\pgayroberts\.ivy2\local\test\test-ui\[revision]\jars
\test-ui.jar

    ==== sonatype-snapshots: tried

      https://oss.sonatype.org/content/repositories/snapshots/test/test-ui/[revi
sion]/test-ui-[revision].pom

      -- artifact test#test-ui;latest.integration!test-ui.jar:

      https://oss.sonatype.org/content/repositories/snapshots/test/test-ui/[revi
sion]/test-ui-[revision].jar

    ==== shared: tried

      C:\Users\pgayroberts\.ivy2\shared\test\test-ui\[revision]\ivy
s\ivy.xml

      -- artifact test#test-ui;latest.integration!test-ui.jar:

      C:\Users\pgayroberts\.ivy2\shared\test\test-ui\[revision]\jar
s\test-ui.jar

    ==== public: tried

      http://repo1.maven.org/maven2/test/test-ui/[revision]/test-ui-[revision].p
om

      -- artifact test#test-ui;latest.integration!test-ui.jar:

      http://repo1.maven.org/maven2/test/test-ui/[revision]/test-ui-[revision].j
ar

        ::::::::::::::::::::::::::::::::::::::::::::::

        ::          UNRESOLVED DEPENDENCIES         ::

        ::::::::::::::::::::::::::::::::::::::::::::::

        :: test#test-ui;latest.integration: not found

        ::::::::::::::::::::::::::::::::::::::::::::::



:: USE VERBOSE OR DEBUG MESSAGE LEVEL FOR MORE DETAILS
synchronizing dependencies in C:\Users\pgayroberts\test\srv\target\
dependency ...
:: retrieving :: test#test [sync]
    confs: [default, runtime, test]
    72 artifacts copied, 0 already retrieved (13694kB/164ms)
DONE
restx> app run
compiling App... [DONE]
copying resources... [DONE]
starting test.AppServer... - type `stop` to stop it and go back to restx shell
2015-03-27 03:19:59,366 [main            ] [          ] INFO  org.eclipse.jetty.
server.Server - jetty-8.1.8.v20121106
LoginService=HashLoginService[null] identityService=org.eclipse.jetty.security.D
efaultIdentityService@2ea227af
2015-03-27 03:20:00,122 [main            ] [          ] INFO  restx.RestxMainRou
terFactory - LOADING MAIN ROUTER
2015-03-27 03:20:00,156 [main            ] [          ] INFO  restx.RestxMainRou
terFactory -
Hot compile is not enabled, no on the fly compilation will be performed
To enable it, use '-Drestx.app.package=<rootAppPackage> -Drestx.router.hotcompil
e=true' as VM argument
and make sure you have JDK tools.jar in your classpath
2015-03-27 03:20:00,157 [main            ] [          ] INFO  restx.RestxMainRou
terFactory -
--------------------------------------
 -- RESTX >> LOAD ON REQUEST << >> DEV MODE << >> HOT RELOAD <<
 -- for admin console,
 --   VISIT http://127.0.0.1:8080/api/@/ui/
 --

2015-03-27 03:20:01,321 [main            ] [          ] INFO  restx.monitor.Metr
icsConfiguration - registering Metrics JVM metrics
2015-03-27 03:20:01,590 [main            ] [          ] INFO  restx.stats.RestxS
tatsCollector - collecting and sharing stats enabled - see http://restx.io/stats
.html for details.
stop
test> exit

C:\Users\pgayroberts>cd whms

C:\Users\pgayroberts\whms>restx
===============================================================================
== WELCOME TO RESTX SHELL - 0.34 - type `help` for help on available commands
===============================================================================
fgwh-ms-parent> app run
can't find manifest.main.classname property in md.restx.json
alternatively you can provide the class to run with `app run <class.to.Run>`
fgwh-ms-parent> ls
.git
.gitconfig
.gitignore
pom.xml
srv
ui
fgwh-ms-parent> cd srv
fgwh-ms> ls
data
ivysettings.xml
logs
md.restx.json
module.ivy
pom.xml
src
target
fgwh-ms> app run
compiling App... [DONE]
copying resources... [DONE]
starting fgwh\ms.AppServer... - type `stop` to stop it and go back to restx shel
l
Error: Could not find or load main class fgwh\ms.AppServer
exit
restx> unrecognized command - type `stop` to stop the app

restx> unrecognized command - type `stop` to stop the app
stop
fgwh-ms> exit

C:\Users\pgayroberts\whms>cd ..

C:\Users\pgayroberts>cd fgwh-ms

C:\Users\pgayroberts\fgwh-ms>ls
fgwh-ms

C:\Users\pgayroberts\fgwh-ms>cd fgwh-ms

C:\Users\pgayroberts\fgwh-ms\fgwh-ms>ls
pom.xml  srv  ui

C:\Users\pgayroberts\fgwh-ms\fgwh-ms>cd srv

C:\Users\pgayroberts\fgwh-ms\fgwh-ms\srv>ls
data  md.restx.json  module.ivy  pom.xml  src  target

C:\Users\pgayroberts\fgwh-ms\fgwh-ms\srv>restx
===============================================================================
== WELCOME TO RESTX SHELL - 0.34 - type `help` for help on available commands
===============================================================================
fgwh-ms> app run
compiling App... [DONE]
copying resources... [DONE]
starting fgwh\ms.AppServer... - type `stop` to stop it and go back to restx shel
l
Error: Could not find or load main class fgwh\ms.AppServer
exit
restx> unrecognized command - type `stop` to stop the app
```

**Comment by xhanin** [comment]
I see several problems here, let me give some explanations.

The first error `test#test-ui;latest.integration: not found` is due to how the dependency on the `ui` (used by maven for war overlay) is handled by Ivy when resolving dependencies. So that dependency is not resolved properly and gives that error. Still it copies the other dependencies so the error is not the source of the problem, it's just that it's not very clean.

Then the first `app run` tells it doesn't find `manifest.main.classname`, this is because you run `app run` from the app root and not in the `srv` directory. This is not properly explained, we should provide at least a better message in this case since the `srv` / `ui` layout is officially supported with `app new`.
 
Finally when you run `app run` in the `srv` directory, it compiles but it doesn't find the main class: this indeed is due to the fact that the `AppServer` class is in a sub directory and this is not properly supported on Windows, the `\` path separator is not properly handled.

So I think we have 3 issues, I'll keep this one for the windows / backslash issue, and open 2 other ones for the other problems.

And thanks for reporting with all these details.

**Comment by fcamblor** [comment]
Once PR #274 will be merged, #188 should be considered as fixed and thus, current issue as well

**Comment by fcamblor** [comment]
Closing issue as #274 has been merged
